lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1406,9 +1406,8 @@ class HSAQueue final : public KalmarQueue
             wait();
         }
         op->asyncOpsIndex(asyncOps.size());
-        asyncOps.push_back(op);
```

Comment:
@scchan why is necessary to move this statement after `op->getCommandKind()`?

Reply:
I changed `push_back` by moving `op` into the vector instead of making a new copy.  I swapped the order so that `op->getCommandKind()` is not referencing an object in an invalid state.